src/main/java/seedu/pill/util/Storage.java

```diff
@@ -79,9 +79,11 @@ public void saveItem(Item item) throws PillException {
         try {
             File file = initializeFile();
             FileWriter fw = new FileWriter(file, true);
-            fw.write((item.getName() + SEPARATOR + item.getQuantity()
-                    + SEPARATOR + item.getExpiryDate())
-                    + System.lineSeparator());
+            fw.write(item.getName() + SEPARATOR + item.getQuantity());
+            if (item.getExpiryDate().isPresent()) {
+                fw.write(SEPARATOR + item.getExpiryDate().get());
+            }
+            fw.write(System.lineSeparator());
```

Comment:
Good that you checked for whether an item is present. Alternatively, you could have use the orElseGet method from Optionals to do this, but this works.